bsms.mediawiki

```diff
@@ -0,0 +1,176 @@
+* The Signer should subsequently use the descriptor to generate and verify receive and change addresses.
+
+This completes the setup.
+
+===Encryption===
+
+====The Token====
+We define three modes of encryption.
+
+# <tt>NO_ENCRYPTION</tt> : the <tt>TOKEN</tt> is set to <tt>0</tt>. Encryption is disabled.
+# <tt>STANDARD</tt> : the <tt>TOKEN</tt> is a 64-bit nonce.
+# <tt>EXTENDED</tt> : the <tt>TOKEN</tt> is a 96-bit nonce.
+
+The <tt>TOKEN</tt> can be converted to one of these formats:
+* A mnemonic phrase using [https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki BIP-0039] word list (6 words in <tt>STANDARD</tt> mode, 9 words in <tt>EXTENDED</tt> mode).
+* A decimal number. The number must not exceed 2^64 in <tt>STANDARD</tt> mode, and 2^96 in <tt>EXTENDED</tt> mode.
+* A QR code.
+* Other formats.
+
+The flexibility in the data format allows each Signer to customize the User Experience based on its respective capabilities.
+
+====Key Derivation====
+The key derivation function is [https://tools.ietf.org/html/rfc2898 PBKDF2], with PRF = SHA512. Specifically:
+
+<tt>DK = PBKDF2(PRF, Password, Salt, c, dkLen)</tt>
+
+Whereas:
+
+* PRF = SHA512
+* Password = "No SPOF"
+* Salt = <tt>TOKEN</tt>
+* c = 2048
+* dkLen = 256
+* DK = Derived <tt>ENCRYPTION_KEY</tt>
+
+====Encryption Scheme====
+The encryption scheme is [https://tools.ietf.org/html/rfc3686 AES-256-CTR].
+
+<tt>Ciphertext = AES-256-CTR-Encrypt(Plaintext, Key, IV)</tt>
+
+<tt>Plaintext = AES-256-CTR-Decrypt(Ciphertext, Key, IV)</tt>
+
+<tt>IV = HMAC-SHA256(HMAC_Key, TOKEN || Data)</tt>
+
+Whereas:
+* Key = <tt>ENCRYPTION_KEY</tt>
+* HMAC_Key = SHA256(<tt>ENCRYPTION_KEY</tt>)
+* Data = the plaintext data, i.e. the entire key record in round 1 and the entire descriptor record in round 2
+
+The IV is to be sent along with the key and descriptor record, as specified above. Because the IV is an HMAC over the entire plaintext, this is essentially an [https://en.wikipedia.org/wiki/Authenticated_encryption#Encrypt-and-MAC_(E&M) Encrypt-and-MAC] form of authenticated encryption.
+
+==QR Codes==
+For signers that use QR codes to transmit data, key and descriptor records can be converted to QR codes, following [https://github.com/BlockchainCommons/Research/blob/master/papers/bcr-2020-005-ur.md the BCR standard].
+
+Also refer to [https://github.com/BlockchainCommons/Research/blob/master/papers/bcr-2020-015-account.md UR Type Definition for BIP44 Accounts] and [https://github.com/BlockchainCommons/Research/blob/master/papers/bcr-2020-010-output-desc.md UR Type Definition for Bitcoin Output Descriptors] for more details.
+
+==Security==
+
+This proposal introduces two layers of protection. The first one is a temporary, secret token, used to encrypt the two rounds of communication between the Signer and the Coordinator. The second one is through the descriptor checksum and visual inspection of the descriptor itself.
```

Comment:
Add something about the new authenticated encryption method used now.

The first one is a temporary, secret token, used to encrypt and authenticate the two rounds of communication...